vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SnapshotManager.java

```diff
@@ -0,0 +1,233 @@
+     * @throws IOException - if an I/O error occurs during the snapshot creation
+     * @throws InterruptedException - if the snapshot creation process is interrupted
+     * @throws FileStorageException - if an error occurs while storing the snapshot
+     * @throws ImageCouldNotBeStoredException 
+     */    
+    @Async
+    @Override
+    @Transactional
+    public void createSnapshot(String resourcesPath, String exhibitionFolderName,SequenceHistory sequenceHistory, ExhibitionSnapshot exhibitionSnapshot) 
+            throws IOException, InterruptedException, FileStorageException {
+        storageEngineDownloads.copyToFolder(exhibitionFolderName + File.separator + RESOURCES_FOLDER_NAME, resourcesPath);
+        List<Space> spaces= spaceRepository.findAllBySpaceStatus(SpaceStatus.PUBLISHED);
+
+        for(Space space : spaces) {
+            renderingManager.downloadSpace(space, exhibitionFolderName, sequenceHistory);                
+        }
+        SnapshotTask snapshotTask = exhibitionSnapshot.getSnapshotTask();
+        snapshotTask.setTaskComplete(true);
+        snapshotTaskRepository.save(snapshotTask);   
+    }
+
+    /**
+     * 
+     * Creates folder for ExhibitionSnapshot and updates entity. 
+     * 
+     * @param exhibitionSnapshot     {@link ExhibitionSnapshot} for which the folder is created
+     * @param exhibitionFolderName   the folder name of the snapshot
+     * @return the name of the exhibition folder
+     */
+    private String createSnapshotFolder(ExhibitionSnapshot exhibitionSnapshot, String exhibitionFolderName) {
+        storageEngineDownloads.createFolder(exhibitionFolderName);
+        exhibitionSnapshot.setFolderName(exhibitionFolderName);
+        exhibitionSnapshotRepository.save(exhibitionSnapshot); 
+        return exhibitionFolderName;
+    }
+
+    /**
+     * Downloads the given exhibition folder by the given ExhibitionSnapshot id
+     * 
+     * @param id  - exhibition snapshot id
+     * @return - The zip file as a byte array
+     * @throws ExhibitionDownloadNotFoundException
+     * @throws IOException
+     */
+    @Override
+    public byte[] getExhibitionSnapshot(String id) throws ExhibitionSnapshotNotFoundException, IOException {
+        Optional<ExhibitionSnapshot> exhibitionSnapshot = exhibitionSnapshotRepository.findById(id);
+
+        if(exhibitionSnapshot.isPresent()) {           
+            try {
+                return storageEngineDownloads.getZip(exhibitionSnapshot.get().getFolderName());                
+            }catch(FileSystemNotFoundException e) {
+                throw new ExhibitionSnapshotNotFoundException(e.getMessage(), e);
+            }
+              
+        } else {
+            throw new ExhibitionSnapshotNotFoundException(id);
+        }
+    }
+
+    /**
+     * Return true is corresponding snapshot is created. Else return false.
+     */
+    @Override
+    @Transactional
+    public Boolean doesSnapshotExist(String id) {
```

Comment:
that is confusing. by snapshot you mean the generated zip file? right now the logic suggests the name should be "isSnapshotCreated" or something like that but then what does it mean if there is no task and the return value is false? It seems like there are to aspects merged into one.